test/suite/test_sweep04.py

```diff
@@ -0,0 +1,225 @@
+                sclose = stat_cursor[stat.conn.dh_session_handles][2]
+                ssweep = stat_cursor[stat.conn.dh_session_sweeps][2]
+                tod = stat_cursor[stat.conn.dh_sweep_tod][2]
+                ref = stat_cursor[stat.conn.dh_sweep_ref][2]
+                dhandles = stat_cursor[stat.conn.dh_conn_handle_count][2]
+                files_open = stat_cursor[stat.conn.file_open][2]
+                self.pr(('DHANDLE STATS: close={}, remove={}, sweep={}, session_handles={}, '+
+                         'session_sweeps={}, sweep_tod={}, sweep_ref={}').format(
+                             close, remove, sweep, sclose, ssweep, tod, ref))
+                self.pr('  dhandle_count={}'.format(dhandles))
+                self.pr('  file_open={}'.format(files_open))
+                dhandle_counts.append(dhandles)
+                stat_cursor.close()
+                if loopcount % 100 == 999:
+                    self.conn.debug_info('handles=true')
+                self.pr('  average,slope={}'.format(str(average_slope(dhandle_counts))))
+
+            # Reset any sessions we've used. This may be necessary to trigger some session sweeps.
+            rand_session.reset()
+            self.session.reset()
+
+        # Half the run is dhandle growth, and the second half should see a decline.
+        # If everything is working right in the first half of the run, we should start
+        # to see the number of dhandles start to asymptote. So we check the slope of
+        # the dhandle line in the second quarter to see that is the case.  In the second
+        # half of the run, where we aren't accessing the transient tables, those references
+        # should free up, and we should see an asymptote at the end much closer to the
+        # number of files.
+        half = len(dhandle_counts)//2
+        qtr = len(dhandle_counts)//4
+        tenth = len(dhandle_counts)//10
+
+        (q1_avg, q1_slope) = average_slope(dhandle_counts[0:qtr])
+        (q2_avg, q2_slope) = average_slope(dhandle_counts[qtr:half])
+        (q3_avg, q3_slope) = average_slope(dhandle_counts[half:-qtr])
+        (q4_avg, q4_slope) = average_slope(dhandle_counts[-qtr:])
+        (end_run_avg, end_run_slope) = average_slope(dhandle_counts[-tenth:])
+
+        self.pr('1st qtr:  average={},slope={}'.format(q1_avg, q1_slope))
+        self.pr('2nd qtr:  average={},slope={}'.format(q2_avg, q2_slope))
+        self.pr('3rd qtr:  average={},slope={}'.format(q3_avg, q3_slope))
+        self.pr('4th qtr:  average={},slope={}'.format(q4_avg, q4_slope))
+
+        self.pr('end run: average={},slope={}'.format(end_run_avg, end_run_slope))
+
+        # Note, we don't check the first half average, it's likely to be big, but its size
+        # depends on many factors. The important thing is that the slope has flattened out.
+        self.assertLess(abs(q2_slope), q1_slope)
+        self.assertLess(abs(q2_slope), 15.0)
```

Comment:
Add to the comment "The important thing..." about how 15.0 means it has flattened out.

Reply:
Added.